qa/tasks/cephfs/test_volumes.py

```diff
@@ -302,6 +305,30 @@ def test_volume_rm(self):
         else:
             raise RuntimeError("expected the 'fs volume rm' command to fail.")
 
+    def test_volume_rm_when_mon_delete_pool_false(self):
+        """
+        That the volume can only be removed when --yes-i-really-mean-it is used
+        and verify that the deleted volume is not listed anymore.
+        """
+        try:
+            self._raw_cmd("config", "set", "mon", "mon_allow_pool_delete", "false")
+            self._fs_cmd("volume", "rm", self.volname, "--yes-i-really-mean-it")
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.EPERM:
+                raise RuntimeError("expected the 'fs volume rm' command to fail with EPERM, "
+                                   "but it failed with {0}".format(ce.exitstatus))
+            else:
+                self._raw_cmd("config", "set", "mon", "mon_allow_pool_delete", "true")
+                self._fs_cmd("volume", "rm", self.volname, "--yes-i-really-mean-it")
+
+                #check if it's gone
+                volumes = json.loads(self._fs_cmd("volume", "ls", "--format=json-pretty"))
+                if (self.volname in [volume['name'] for volume in volumes]):
+                    raise RuntimeError("Expected the 'fs volume rm' command to succeed. "
+                                       "The volume {0} not removed.".format(self.volname))
```

Comment:
Also check if the pools are gone too?

Reply:
I added the tests but it's failing because of another bug. I have created the tracker https://tracker.ceph.com/issues/45910 and will send the patch.

Reply:
I have posted the PR https://github.com/ceph/ceph/pull/35420 for the same. I will rebase this once that is merged.

Reply:
Done